### PR TITLE
[3/3] loopd: add macaroon bakery

### PR DESCRIPTION
**Pull request by guggero**
Split off from https://github.com/lightninglabs/loop/pull/281, only the last 5 commits are new.

To give users the ability to fully customize their access with custom macaroons, the two functionalities `BakeMacaroon` and `ListPermissions` are copied from `lnd` and added to both the RPC and command line.

### CODE REVIEW

**Review by joostjager — COMMENTED** [review]
Please re-request review when 0.11.1 has been released.

File: Makefile [hunk 1]
```diff
@@ -58,6 +58,10 @@ mod-check:
 	$(GOMOD) tidy
 	if test -n "$$(git status | grep -e "go.mod\|go.sum")"; then echo "Running go mod tidy changes go.mod/go.sum"; git status; git diff; exit 1; fi
 
+rpc:
```
Comment (carlaKC) [inline Makefile:61]: 🎉

File: looprpc/client.proto [hunk 2]
```diff
@@ -699,3 +710,19 @@ message BakeMacaroonResponse {
...
+message ListPermissionsRequest {
+}
+
+message ListPermissionsResponse {
+    /*
+    A map between all RPC method URIs and their required macaroon permissions to
+    access them.
+    */
+    map<string, MacaroonPermissionList> method_permissions = 1;
+}
```
Comment (carlaKC) [inline looprpc/client.proto:728]: nit: newline

**Review by carlaKC — COMMENTED** [review]
Pretty straightforward since this is mostly copy-paste. Tested out baking + using a macaroon, works 🍪 

Added this comment in the [1/2] PR, but it would be nice to update the `README` at the same time. LGTM, but waiting on `0.11.1` and the first PR before approval.

**Comment by Roasbeef** [comment]
Is this still relevant or did an alternative land? (given `loopd` is now able to work within `litd`).

**Comment by guggero** [comment]
Closing this as there currently doesn't seem to be a real need for baking custom Loop macaroons. Might pick this up again later.

**Pull request closed without merging**